framework/src/systems/NonlinearSystemBase.C

```diff
@@ -3243,7 +3251,7 @@ NonlinearSystemBase::computeScalingJacobian(NonlinearImplicitSystem & sys)
       displaced_problem->systemBaseNonlinear().applyScalingFactors(inverse_scaling_factors);
 
     // Now it's essential that we reset the sparsity pattern of the matrix
-    auto ierr = MatResetPreallocation(petsc_matrix.mat());
+    ierr = MatResetPreallocation(petsc_matrix.mat());
```

Comment:
I was wondering if it is possible to move `MatResetPreallocation`  and `MatGetLocalSize` to PetscSupport.C?


I am fine that  we have a lot of PETSc code in MOOSE, which will secure my job:-)  But other team members may try to keep things in one single place, and it is easier  to maintain.


Or add interfaces to `libMesh`. Up to you!

Reply:
Sigh, I will make the interfaces in libMesh. What a ruthless reviewer! 😄

Reply:
I am trying to help you make your code even more beautiful.